packages/flutter_tools/lib/src/macos/build_macos.dart

```diff
@@ -108,6 +109,32 @@ Future<void> buildMacOS({
   if (!flutterBuildDir.existsSync()) {
     flutterBuildDir.createSync(recursive: true);
   }
+
+  final Directory xcodeProject = flutterProject.macos.xcodeProject;
+
+  // If the standard project exists, specify it to getInfo to handle the case where there are
+  // other Xcode projects in the macos/ directory. Otherwise pass no name, which will work
+  // regardless of the project name so long as there is exactly one project.
+  final String? xcodeProjectName = xcodeProject.existsSync() ? xcodeProject.basename : null;
+  final XcodeProjectInfo? projectInfo = await globals.xcodeProjectInterpreter?.getInfo(
+    xcodeProject.parent.path,
+    projectFilename: xcodeProjectName,
+  );
+  final String? scheme = projectInfo?.schemeFor(buildInfo);
+  if (scheme == null) {
+    projectInfo!.reportFlavorNotFoundAndExit();
+  }
+  final String? configuration = projectInfo?.buildConfigurationFor(buildInfo, scheme);
+  if (configuration == null) {
+    throwToolExit('Unable to find expected configuration in Xcode project.');
+  }
+
+  final Map<String, String> buildSettings = await flutterProject.macos.buildSettingsForBuildInfo(
```

Comment:
This code is new, it's running the formerly iOS-only function wrapping `xcodebuild -showBuildSettings` command and populating a map with the results.